libraries/rush-lib/src/logic/operations/OperationMetadataManager.ts

```diff
@@ -121,11 +130,13 @@ export class OperationMetadataManager {
   }): Promise<void> {
     await this.stateFile.tryRestoreAsync();
 
-    // Append cached log into current log file
-    terminal.writeLine(`Restoring cached log file at ${this._logPath}`);
+    const operationSucceeded: boolean = !(
+      this.stateFile.state?.status && RESTORE_FROM_ERROR_STATUSES.has(this.stateFile.state?.status)
+    );
+
     let logReadStream: fs.ReadStream | undefined;
     try {
-      logReadStream = fs.createReadStream(this._logPath, {
+      logReadStream = fs.createReadStream(operationSucceeded ? this._logPath : this._errorLogPath, {
         encoding: 'utf-8'
       });
       for await (const data of logReadStream) {
```

Comment:
If you are running with `--verbose`, this will produce different log output compared to a local build. To fully replay you would need to create a log file format that just the raw series of chunks with whether they are stdout or stderr.

Reply:
Don't we have that already with the `all.log` and `error.log`? For errors/warnings, couldn't we just write `all.log` to the buildCacheTerminal and `error.log` to the terminal from `runWithTerminalAsync`? I think that would keep the `--verbose` output while still reflecting log items in the operation summary view.

Reply:
I think I see the issue

### main build agent
```
==[ b (build) ]===================================================[ 1 of 10 ]==

This project was not found in the build cache.
Invoking: node ../build.js b 
start b
done b
Caching build output folders: dist
Trying to find "tar" binary
Successfully set cache entry.
Cache key: eccd46a82deac50ce7e2b4f80638862a5b76f60a-foo-warnings
"b (build)" completed with warnings in 2.09 seconds.
```

### restore from cache agent
```
==[ b (build) ]===================================================[ 1 of 10 ]==

This project was not found in the build cache.
Build cache hit.
Cache key: eccd46a82deac50ce7e2b4f80638862a5b76f60a-foo-warnings
Clearing cached folders: dist
Successfully restored output from the build cache.
done b
Invoking: node ../build.js b 
start b
done b
"b (build)" completed with warnings in 0.02 seconds.
```

Reply:
@dmichon-msft I updated it to a writable that stores an intermediate `log-chunks.json` file, can you take a look? Example build cache output,
```
Build cache hit.
Cache key: fc2841bb73a27572ce0f5311a172a3f2f1106fb5-foo-warnings
Clearing cached folders: dist
Successfully restored output from the build cache.
Invoking: node ../build.js a 
start a
done a
"a (build)" completed with warnings in 0.01 seconds.
```

and the summary view,
```
--[ WARNING: a (build) ]-------------------------------------[ 0.01 seconds ]--

done a
```